### PR TITLE
Move ResetMetricsOnDelete to Beta

### LINKED ISSUE #3100: Graduate `ResetMetricsOnDelete` to Beta
I propose that we graduate the `ResetMetricsOnDelete` feature to beta. It has been in alpha since 1.26.0. 

Context:
https://development.agones.dev/site/docs/guides/feature-stages/#feature-gates

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

/kind feature

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3100

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Guides/feature-stages.md [hunk 1]
```diff
@@ -49,9 +49,9 @@ The current set of `alpha` and `beta` feature gates:
 | [GameServer Stable Network ID]({{% ref "/docs/Reference/gameserver.md#stable-network-id" %}})                         | `PodHostname`                  | Enabled  | `Beta`  | 1.32.0 |
 | [Graceful Termination for GameServer SDK](https://github.com/googleforgames/agones/pull/2205)                         | `SDKGracefulTermination`       | Enabled  | `Beta`  | 1.18.0 |
 | [GameServer state filtering on GameServerAllocations](https://github.com/googleforgames/agones/issues/1239)           | `StateAllocationFilter`        | Enabled  | `Beta`  | 1.26.0 |
+| [Reset Metric Export on Fleet / Autoscaler deletion]({{% relref "./metrics.md#dropping-metric-labels" %}})            | `ResetMetricsOnDelete`         | Enabled  | `Beta`  | 1.26.0 |
```
Comment (zmerlynn) [inline site/content/en/docs/Guides/feature-stages.md:52]: nit: was trying to keep this list sorted by `Stage`, then the name of the gate

File: pkg/util/runtime/features.go [hunk 2]
```diff
@@ -110,11 +110,11 @@ var (
 		FeatureSDKGracefulTermination: true,
 		FeatureStateAllocationFilter:  true,
 		FeaturePodHostname:            true,
+		FeatureResetMetricsOnDelete:   true,
```
Comment (zmerlynn) [inline pkg/util/runtime/features.go:113]: can you fix the ordering here as well? It should be alphabetical within the stage. Looks like `PodHostname` also violates right now.

File: install/helm/agones/defaultfeaturegates.yaml [hunk 3]
```diff
@@ -19,11 +19,11 @@ CustomFasSyncInterval: true
 SDKGracefulTermination: true
 StateAllocationFilter: true
 PodHostname: true
+ResetMetricsOnDelete: true
```
Comment (zmerlynn) [inline install/helm/agones/defaultfeaturegates.yaml:22]: can you fix the ordering here? It should be alphabetical within the stage. Looks like `PodHostname` also violates right now.

File: site/content/en/docs/Guides/feature-stages.md (hunk 1 above)
Comment (Kalaiselvi84) [reply site/content/en/docs/Guides/feature-stages.md]: I have rearranged the features in alphabetical order.

File: pkg/util/runtime/features.go (hunk 2 above)
Comment (Kalaiselvi84) [reply pkg/util/runtime/features.go:112]: rearranged the order

File: install/helm/agones/defaultfeaturegates.yaml (hunk 3 above)
Comment (Kalaiselvi84) [reply install/helm/agones/defaultfeaturegates.yaml:20]: rearranged it

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**